tools/superdsm/superdsm.xml

```diff
@@ -91,14 +128,37 @@
         </section>
     </inputs>
     <outputs>
-        <data format="json" name="cfg" from_work_dir="cfg.json" label="${tool.name} on ${on_string}: cfg" />
-        <data format="png" name="masks" from_work_dir="masks.png" label="${tool.name} on ${on_string}: masks" />
-        <data format="png" name="overlay" from_work_dir="overlay.png" label="${tool.name} on ${on_string}: overlay" />
+        <data format="png" name="masks" from_work_dir="masks.png" label="${tool.name} on ${on_string}: masks">
+            <filter>outputs['do_masks_enabled'] == 'true'</filter>
+        </data>
+        <data format="tsv" name="cfg" from_work_dir="cfg.tsv" label="${tool.name} on ${on_string}: cfg">
+            <filter>outputs['do_cfg_enabled'] == 'true'</filter>
+        </data>
+        <data format="png" name="overlay" from_work_dir="overlay.png" label="${tool.name} on ${on_string}: overlay">
+            <filter>outputs['do_overlay']['do_overlay_enabled'] == 'true'</filter>
+        </data>
     </outputs>
     <tests>
-        <test>
+        <test expect_num_outputs="3">
             <param name="dataset" value="BBBC033_C2_z28.png" />
+            <param name="do_masks_enabled" value="true" />
+            <param name="do_overlay_enabled" value="true" />
+            <param name="do_cfg_enabled" value="true" />
             <output name="overlay" value="overlay.png" ftype="png" compare="sim_size" />
+            <output name="cfg" value="cfg.tsv" ftype="tsv" compare="sim_size" />
+        </test>
+        <test expect_num_outputs="1">
+            <param name="dataset" value="BBBC033_C2_z28.png" />
+            <param name="do_masks_enabled" value="false" />
+            <param name="do_overlay_enabled" value="true" />
+            <param name="do_cfg_enabled" value="false" />
+            <output name="overlay" value="overlay.png" ftype="png" compare="sim_size" />
+        </test>
+        <test expect_num_outputs="0">

```

Comment:
I don't think a tool with 0 outputs is valid or we should have that.

Reply:
True, a tool producing zero outputs makes no sense. But is it possible to add a constraint to the tool, that at least one of the three `select` input fields must be set to `true`? I haven't found that (yet)…